gridcomps/History/MAPL_HistoryCollection.F90

```diff
@@ -11,21 +11,33 @@ module MAPL_HistoryCollectionMod
   use HistoryTrajectoryMod
   use gFTL_StringStringMap
   implicit none
-  
+
   private
 
   type, public :: FieldSet
      character(len=ESMF_MAXSTR), pointer :: fields(:,:) => null()
      integer :: nfields = 0
   end type FieldSet
 
+  type, public :: HistoryCollectionGlobalAttributes
+     character(len=ESMF_MAXSTR) :: filename
+     character(len=ESMF_MAXSTR) :: descr
+     character(len=ESMF_MAXSTR) :: comment
+     character(len=ESMF_MAXSTR) :: contact
+     character(len=ESMF_MAXSTR) :: convention
+     character(len=ESMF_MAXSTR) :: institution
+     character(len=ESMF_MAXSTR) :: references
+     character(len=ESMF_MAXSTR) :: source
+     contains
+        procedure :: define_collection_attributes
+  end type HistoryCollectionGlobalAttributes
+
```

Comment:
I want to suggest the following change, but ... won't work with the subsequent use in ESMF.  Sigh.

```suggestion
  type, public :: HistoryCollectionGlobalAttributes
     character(:), allocatable :: filename
     character(:), allocatable :: descr
     character(:), allocatable :: comment
     character(:), allocatable :: contact
     character(:), allocatable :: convention
     character(:), allocatable :: institution
     character(:), allocatable :: references
     character(:), allocatable :: source
     contains
        procedure :: define_collection_attributes
  end type HistoryCollectionGlobalAttributes

```